tests/python/pants_test/pantsd/test_pantsd_integration.py

```diff
@@ -138,66 +133,56 @@ def test_pantsd_run(self):
+      run = self.run_pants_with_workdir('help | head -1', workdir, pantsd_config, shell=True)
+      self.assertNotIn('broken pipe', run.stderr_data.lower())
+      checker.await_pantsd()
 
   def test_pantsd_stacktrace_dump(self):
-    with self.pantsd_test_context() as (workdir, pantsd_config, checker):
-      print('log: {}/pantsd/pantsd.log'.format(workdir))
-      # Explicitly kill any running pantsd instances for the current buildroot.
-      self.assert_success(self.run_pants_with_workdir(['kill-pantsd'], workdir, pantsd_config))
-      try:
-        # Start pantsd implicitly via a throwaway invocation.
-        self.assert_success(self.run_pants_with_workdir(['help'], workdir, pantsd_config))
-        checker.await_pantsd()
+    with self.pantsd_successful_run_context() as (pantsd_run, checker, workdir):
+      pantsd_run(['help'])
+      checker.await_pantsd()
 
-        os.kill(checker.pid, signal.SIGUSR2)
+      os.kill(checker.pid, signal.SIGUSR2)
 
-        # Wait for log flush.
-        time.sleep(2)
+      # Wait for log flush.
+      time.sleep(2)
 
-        self.assertIn('Current thread 0x', '\n'.join(read_pantsd_log(workdir)))
-      finally:
-        # Explicitly kill pantsd (from a pantsd-launched runner).
-        self.assert_success(self.run_pants_with_workdir(['kill-pantsd'], workdir, pantsd_config))
-        checker.assert_stopped()
+      self.assertIn('Current thread 0x', '\n'.join(read_pantsd_log(workdir)))
 
-  def test_pantsd_runner_doesnt_die_after_failed_run(self):
+  def test_pantsd_pantsd_runner_doesnt_die_after_failed_run(self):
     with self.pantsd_test_context() as (workdir, pantsd_config, checker):
-      self.assert_success(self.run_pants_with_workdir(['kill-pantsd'], workdir, pantsd_config))
-      try:
-        # Start pantsd implicitly via a throwaway invocation.
-        self.assert_success(self.run_pants_with_workdir(['help'], workdir, pantsd_config))
-        checker.await_pantsd()
-
-        # Run target that throws an exception in pants.
-        self.assert_failure(
-          self.run_pants_with_workdir(
-            ['bundle', 'testprojects/src/java/org/pantsbuild/testproject/bundle:missing-files'],
-            workdir,
-            pantsd_config)
-        )
-        checker.assert_running()
-
-        # Check for no stray pantsd-runner prcesses.
-        self.assertFalse(check_process_exists_by_command('pantsd-runner'))
-
-        # Assert pantsd is in a good functional state.
-        self.assert_success(self.run_pants_with_workdir(['help'], workdir, pantsd_config))
+      # Run target that throws an exception in pants.
+      self.assert_failure(
+        self.run_pants_with_workdir(
+          ['bundle', 'testprojects/src/java/org/pantsbuild/testproject/bundle:missing-files'],
+          workdir,
+          pantsd_config)
+      )
+      checker.await_pantsd()
+
+      # Check for no stray pantsd-runner prcesses.
+      self.assertFalse(check_process_exists_by_command('pantsd-runner'))
```

Comment:
There used to be a `checker.assert_running()` here which feels important

Reply:
`checker.await_pantsd()` implicitly calls `assert_running()`, so it should be equivalent.